### PR TITLE
Simplify client creation and config loading

### PR DESCRIPTION
**Pull request by sethvargo**
This PR does a few things. I can try and break it up if that's easier, but many of the changes are tightly coupled. The goal of this PR is to actually loosen those couplings.

1. Remove the construction of installID files from `InstallInfo` and put it on the client instead. `InstallInfo` really shouldn't have known about constructing filepaths. The pattern of passing in an "override file" was a smell.

1. Remove the embedded MetricsConfig struct in the Client (just copy the required `serverURL` field).

1. Move MetricsConfig validation onto itself (it owns validation, not the Client).

1. Remove the interim "options" option and use the `Client` directly. This removes many layers of indirection. Storing things like the lookuper and nowFunc directly on the `Client` also dramatically simplify initialization and reduces the number of "things" we need to pass around. Instead of "installFiles" being an override, they are just a normal part of the API (with the default value being the default path that was previously constructed in `InstallInfo`.

1. Use `time.Format(time.RFC3339Nano)` instead of `MarshalText`; yes `MarshalText` handles an out-of-range date, but that edge case isn't worth complicating the code. We could use `time.RFC3339` (or any other representation instead), but `RFC3339Nano` is what `MarhsalText` uses internally.

### CODE REVIEW

**Review by drevell — APPROVED** [review]

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -45,72 +48,101 @@ const (
 // is a Client with optOut = true.
 //
 // It is initialized once when called the first time.
-var noopWriterOnce = sync.OnceValue[*Client](func() *Client {
+var noopWriterOnce = sync.OnceValue(func() *Client {
```
Comment (pdewilde) [inline pkg/metrics/metrics.go:51]: Why remove the generic?

Does the compiler implicitly know the value so its redundant?

File: pkg/metrics/metrics_test.go [hunk 2]
```diff
@@ -105,17 +103,18 @@ func TestNew(t *testing.T) {
 
 				installPath := t.TempDir() + "/" + installTimeFileName
 				if tc.installTime != "" {
-					if err := storeInstallTime(testAppID, installPath, &installInfo{tc.installTime}); err != nil {
-						t.Fatalf("test setup failed: %s", err.Error())
+					if err := storeInstallInfo(installPath, &InstallInfo{
+						InstallTime: tc.installTime,
+					}); err != nil {
+						t.Fatal(err)
```
Comment (pdewilde) [inline pkg/metrics/metrics_test.go:109]: Is cuddling errors not seen as necessasry in test code?

Comment (sethvargo) [reply pkg/metrics/metrics_test.go:109]: wdym? this error is cuddled.

Comment (pdewilde) [reply pkg/metrics/metrics_test.go:109]: Oops, wrong term. I meant wrapped with an error and description.

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (pdewilde) [reply pkg/metrics/metrics.go:51]: Interesting, didn't realize that was a feature, assumed you would get an `any` back and need to cast.

File: pkg/metrics/metrics_test.go (hunk 2 above)
Comment (sethvargo) [reply pkg/metrics/metrics_test.go:109]: Ah okay:

- "cuddling" is the `if err := thing; err != nil {}` pattern

- "wrapping" is adding extra context, usually with `%w`

We differ from the internal style guide a bit, but we don't usually add extra context to test failures unless it cannot be inferred from the test name and stacktrace

**Pull request merged**